samcli/lib/package/utils.py

```diff
@@ -166,9 +168,14 @@ def upload_local_artifacts(
 
     local_path = make_abs_path(parent_dir, local_path)
 
+    compression_type = zipfile.ZIP_DEFLATED
+    if no_compression[0]:
```

Comment:
Why is this condition checking for the 0 index of no_compression? It's typed as a bool in the function definition but turned into a tuple earlier in the control flow. I'm not completely following the logic here.

Reply:
Turning it into a tuple was unintentional, wasn't sure how to make it a bool instead. =(